AmplifyPlugins/Auth/Tests/AWSCognitoAuthPluginUnitTests/TaskTests/ClientBehaviorTests/SignIn/AWSAuthSignInWithOTPTaskTests.swift

```diff
@@ -532,31 +559,504 @@ class AWSAuthSignInWithOTPTaskTests: BasePluginTest {
+    ///
+    /// - Given: Given an auth plugin with mocked service and a in progress signIn waiting for SMS verification
+    ///
+    /// - When:
+    ///    - I invoke another signIn with valid values
+    /// - Then:
+    ///    - I should get a .done response
+    ///
+    func testRestartSignIn() async {
+        setUpWith(authPasswordlessEnvironment:
+                    BasicPasswordlessEnvironment(authPasswordlessFactory: makeAuthPasswordlessClient))
+        self.mockIdentityProvider = MockIdentityProvider(mockInitiateAuthResponse: { _ in
+            return InitiateAuthOutput(
+                authenticationResult: .none,
+                challengeName: .customChallenge,
+                challengeParameters: [
+                    "nextStep": "PROVIDE_AUTH_PARAMETERS"
+                ],
+                session: "someSession")
+        }, mockRespondToAuthChallengeResponse: { _ in
+            return RespondToAuthChallengeOutput(
+                authenticationResult: .none,
+                challengeName: .customChallenge,
+                challengeParameters: [
+                    "nextStep": "PROVIDE_CHALLENGE_RESPONSE",
+                    "attributeName": "email",
+                    "deliveryMedium": "EMAIL",
+                    "destination": "S***@g***"
+                ],
+                session: "session")
+        })
+
+        let options = AuthSignInWithOTPRequest.Options()
+        do {
+            let result = try await plugin.signInWithOTP(
+                username: "username",
+                flow: .signIn,
+                destination: .email,
+                options: options)
+            guard case .confirmSignInWithOTP =  result.nextStep else {
+                XCTFail("Result should be .confirmSignInWithOTP for next step instead got. \(result.nextStep)")
+                return
+            }
+            XCTAssertFalse(result.isSignedIn)
+            self.mockIdentityProvider = MockIdentityProvider(mockInitiateAuthResponse: { _ in
+                InitiateAuthOutput(
+                    authenticationResult: .none,
+                    challengeName: .customChallenge,
+                    challengeParameters: [
+                        "nextStep": "PROVIDE_AUTH_PARAMETERS"
+                    ],
+                    session: "someSession")
+            }, mockRespondToAuthChallengeResponse: { _ in
+                RespondToAuthChallengeOutput(
+                    authenticationResult: .init(
+                        accessToken: Defaults.validAccessToken,
+                        expiresIn: 300,
+                        idToken: "idToken",
+                        newDeviceMetadata: nil,
+                        refreshToken: "refreshToken",
+                        tokenType: ""),
+                    challengeName: .none,
+                    challengeParameters: [:],
+                    session: "session")
+            })
+            let result2 = try await plugin.signInWithOTP(
```

Comment:
Didn't understand this - if the same user calls signInWithOTP twice - it'd sign the user in in the second call without calling confirmSignInWithOTP?